packages/workbox-background-sync/Queue.mjs

```diff
@@ -132,17 +134,17 @@ class Queue {
       replayedRequests.push(replay);
     }
 
+    await this._runCallback('queueDidReplay', replayedRequests);
+
     // If any requests failed, put the failed requests back in the queue
-    // and register for another sync.
+    // and reject promise.
     if (failedRequests.length) {
       await Promise.all(failedRequests.map((storableRequest) => {
         return this._queueStore.addEntry(storableRequest);
       }));
 
-      await this._registerSync();
+      return Promise.reject(failedRequests);
```

Comment:
Nit: can we throw here instead of returning a rejected promise? I think it's better to use async function patterns when inside an async function (rather that promise patterns).

Reply:
so `throw new Error(failedRequests);`?

Reply:
I was thinking just:

```js
throw failedRequests;
```

In JS you can throw any value.

cc: @gauntface and @jeffposnick to see if they have thoughts on this.

In general I think we should stick to the pattern that async function use `await`, `try`, `catch`, and `throw`, and regular functions use `resolve` and `reject`, though I guess this may be a bit of a special case...

Reply:
Does anything end up consuming the rejected promise that's returned by this function? Or is the fact that it's a rejected promise all that matters, with the rejection value ignored?

If the value is ignored, then I'd rather see

```
throw new Error('<some meaningful message>');
```

so that a developer reading the source has a bit more context. And if at some point in the future logging is added in, it will end up with a meaningful log message.

If the value isn't ignored, and it needs to be that array, then using `Promise.reject()` seems fine to me, but I'd ask that you add in a comment explaining what's going on.

Reply:
yeah nothing really consumes it.

+ i was thinking something like`throw new Error('X number of requests failed');`

Reply:
Please use a WorkboxError.

Reply:
Done